raster/src/main/scala/geotrellis/raster/imagery/CloudRemoval.scala

```diff
@@ -0,0 +1,57 @@
+package geotrellis.raster.imagery
+
+import geotrellis.raster._
+import geotrellis.raster.render._
+import geotrellis.raster.io.geotiff.SingleBandGeoTiff
+
+import java.io.File
+import spire.syntax.cfor._
+
+object CloudRemoval {
+
+  def cloudRemovalSingleBand(images: Array[Tile], threshold: Int) : Tile = {
+    val headImage = images(0)
+    val result = ArrayTile.empty(headImage.cellType, headImage.cols, headImage.rows)
+
+    cfor(0)(_ < result.rows, _ + 1) { row =>
+      cfor(0)(_ < result.cols, _ + 1) { col =>
+        var sum = 0
+        var count = 0
+        cfor(0)(_ < images.length, _ + 1) { i =>
+          val v = images(i).get(col, row)
+          if(isData(v) && v < threshold) {
+            sum += v
+            count += 1
+          }
+        }
+        
+        result.set(col, row, sum / count)
```

Comment:
Since this is just using a central tendency, is there a good reason to use the mean rather than the median? With a small number of photos (or a large proportion of cloud covered images), I imagine this will cause some washing out of actual values.